zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLQueryParser.scala

```diff
@@ -77,13 +77,18 @@ class AQLQueryParser extends IQueryParser {
     val (group, varMapAfterGroup) = query.groups.map(parseGroupby(_, varMapAfterUnnest, groupVar))
       .getOrElse(("", varMapAfterUnnest))
 
+
     val varName = if (group.length > 0) groupVar else sourceVar
     val (selectPrefix, select) = query.select.map(parseSelect(_, varMapAfterGroup, group.length > 0, varName))
       .getOrElse("", "")
 
     val returnStat = if (query.groups.isEmpty && query.select.isEmpty) s"return $sourceVar" else ""
 
-    Seq(selectPrefix, dataset, lookup, filter, unnest, group, select, returnStat).mkString("\n")
+    val aggrVar = "$c"
+    val aggrVarName = if(selectPrefix.length > 0 ) aggrVar else groupVar
+    val (globalAggrPrefix, aggrReturnStat) = query.globalAggr.map(parseGlobalAggr(_, varMapAfterGroup, aggrVarName)).getOrElse("","")
```

Comment:
like we discussed in the morning, the `varMapAfterGroup` may not be the last varmap, it could have some selection afterwards, so we need to generate the `varMapAfterSelect` and decide which varMap should be used here.